elide-async/src/main/java/com/yahoo/elide/async/service/DefaultAsyncQueryDAO.java

```diff
@@ -195,4 +199,32 @@ protected Object executeInTransaction(DataStore dataStore, Transactional action)
         }
         return result;
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public Collection<AsyncQuery> getActiveAsyncQueryCollection() {
+        Collection<AsyncQuery> asyncQueryList = null;
+
+        log.debug("getActiveAsyncQueryCollection");
+        String filterExpression = "status=in=(" + QueryStatus.CANCELLED.toString() + ","
```

Comment:
we should expand this where clause to pull records based on create_time also as after sometime CANCELLED query list will expand. Something like create_time after current-date minus maxRunTimeSeconds;

Reply:
Also the FilterExpression should be the method argument as elide will be controlling the Filter, where as users can provide their implementation of getActiveAsyncQueryCollection method.